src/cobra/core/gene.py

```diff
@@ -299,3 +483,160 @@ def _repr_html_(self):
             n_reactions=len(self.reactions),
             reactions=format_long_string(", ".join(r.id for r in self.reactions), 200),
         )
+
+
+class GPR(Module):
+    """A Gene Reaction rule in a cobra model, using AST as base class
+
+    Parameters
+    ----------
+    string_gpr : string
+        A GPR in string format, which will be parsed into AST
+    """
+
+    def __init__(self, string_gpr=None, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._genes = set()
+        if string_gpr:
+            self.from_string(string_gpr)
+
+    def from_string(self, string_gpr) -> None:
```

Comment:
Will leave parse_gpr with a deprecation warning, since there still might be code using it. I think we should give people a few minor versions before removing it.

Reply:
Agreed, I didn't mean to remove it. Just to implement it only once and have the other version just call the first implementation. See above.